lib/pure/httpclient.nim

```diff
@@ -1166,9 +1167,18 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   var lastURL = url
   for i in 1..client.maxRedirects:
     if result.status.redirection():
+      if hasKey(result.headers, "Connection"):
+        if result.headers["Connection"] == "close":
+          client.close()
+      if not client.getBody and client.connected:
+        # downloadFile sets client.getBody=false
+        # but we must read the body of redirect responses
+        await parseBody(client, result.headers, result.version)
```

Comment:
This isn't correct. You cannot parse the body here, especially considering that the request might be a HEAD request. Also, reading the body when ``getBody`` is false is incredibly misleading. This seems like a workaround to me.

Reply:
We are in the redirection handling here. It doesn't matter what kind of request was sent before.
The server we contacted replied with a redirection response which usually includes some data besides the http headers. I am assuming that before you can send data on the same connection (to send the request to the new location which might be on the same server) you have to read all data available to read on the socket. 

I fixed the test failures on the windows side. The failure of the last test isn't related to httpclient.

Reply:
Does this only fix the ``downloadFile`` problem? If so then I definitely cannot accept it. We need to fix ``downloadFile`` instead of adding workarounds here for the issues that ``downloadFile`` creates.